### PR TITLE
Initial real commit lol

### PR DESCRIPTION
**Pull request by AlanAyy**
Made a functional EmberBot using Discord.py

Current Admin commands (user invoking the command must have Administrator permissions within any of their roles):
- `!group new {name} {#}`
- `!group list` (Unavailable)
- `!group delete` (Unavailable)
- `!group clear` (Unavailable)
Name MUST include the wildcard character if you want to make more than one channel (currently `?`)
Ex: `!group new saturday-? 22` will make 22 text + voice channels and roles from `saturday-1` to `saturday-22`

Current user commands:
- `!group set {name}`
Name must match an existing channel AND role to be given out (will not match with key roles like Exec, Mentor or President no matter what)
- `!roll 2d6`

### CODE REVIEW

**Comment by AlanAyy** [comment]
Warning: Admin line 74 and 77
Replace `(embed)` with `(embed=embed)`

File: cogs/admin.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        await ctx.guild.create_voice_channel(name, overwrites=permissions, category=category)
+        return True
+        # Use the following if the permissions dict isn't working
+        #   channel.set_permissions(ctx.guild.default_role, read_messages=False, send_messages=False)
+        #   channel.set_permissions(role, read_messages=True, send_messages=True)
+
+    async def set(self, ctx, group):
+        embed = discord.Embed(color=discord.Colour.orange())
+        for channel in ctx.guild.channels:
+            if group == channel.name and group not in ['Exec', 'President', 'Mentor']:
```
Comment (AlphaRLee) [inline cogs/admin.py:71]: Guard against dumb uppercase/lowercase mistakes, make it case insensitive
```suggestion
            if group == channel.name and group.lower() not in ['exec', 'president', 'mentor']:
```

File: cogs/admin.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                # It will attempt to give them the role with the same channel name
+                embed.add_field(name='Role added!', value=group)
+                await ctx.send(embed)
+                return await ctx.author.add_roles(discord.utils.get(ctx.guild.roles, name=group))
+        embed.add_field(name='Role could not be added!', value=group)
+        return await ctx.send(embed)
+
+    async def delete(self, ctx, name):
+        pass
```
Comment (AlphaRLee) [inline cogs/admin.py:80]: Replace with a helpful error message
```suggestion
        ctx.send("Sorry, this is not implemented yet")
```

File: cogs/admin.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    async def clear(self, *args):
```
Comment (AlphaRLee) [inline cogs/admin.py:82]: Replace with a helpful error message
```suggestion
        ctx.send("Sorry, this is not implemented yet")
```

File: cogs/admin.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        pass
```
Comment (AlphaRLee) [inline cogs/admin.py:83]: Replace with a helpful error message
```suggestion
        ctx.send("Sorry, this is not implemented yet")
```

File: cogs/admin.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    async def set(self, ctx, group):
+        embed = discord.Embed(color=discord.Colour.orange())
+        for channel in ctx.guild.channels:
+            if group == channel.name and group not in ['Exec', 'President', 'Mentor']:
+                # It will attempt to give them the role with the same channel name
+                embed.add_field(name='Role added!', value=group)
+                await ctx.send(embed)
+                return await ctx.author.add_roles(discord.utils.get(ctx.guild.roles, name=group))
+        embed.add_field(name='Role could not be added!', value=group)
+        return await ctx.send(embed)
```
Comment (AlphaRLee) [inline cogs/admin.py:77]:
```suggestion
        return await ctx.send(embed=embed)
```

File: cogs/admin.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        #   channel.set_permissions(ctx.guild.default_role, read_messages=False, send_messages=False)
+        #   channel.set_permissions(role, read_messages=True, send_messages=True)
+
+    async def set(self, ctx, group):
+        embed = discord.Embed(color=discord.Colour.orange())
+        for channel in ctx.guild.channels:
+            if group == channel.name and group not in ['Exec', 'President', 'Mentor']:
+                # It will attempt to give them the role with the same channel name
+                embed.add_field(name='Role added!', value=group)
+                await ctx.send(embed)
```
Comment (AlphaRLee) [inline cogs/admin.py:74]:
```suggestion
                await ctx.send(embed=embed)
```

File: cogs/admin.py (hunk 1 above)
Comment (AlphaRLee) [reply cogs/admin.py:71]: Actually, might be nice to put this in a config file in the future...

File: startup.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+from discord.ext import commands
+
+from cogs.utils.config import *
+
+
+# First time run config
+def wizard():
+    config = {}
+    print('Welcome to AlanAyy\'s Ignite-Ember setup!')
```
Comment (AlphaRLee) [inline startup.py:12]: Would you be ok taking your name off the print? Preferably this belongs to Schulich Ignite. Your name defs belongs on a `contributors.md` file tho :)
```suggestion
    print('Welcome to Ignite-Ember setup!')
```

File: startup.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    with open('settings/config.json', encoding='utf-8', mode='r') as fp:
+        data = json.load(fp)  # Let's see if we've got a valid settings file
+except IOError:
+    wizard()
+
+# Now that all the setup's finished, we're good to go!
+logging.basicConfig(level=logging.INFO)
+bot = commands.Bot(
+    command_prefix=get_config_value('config', 'prefix'),
+    description='''Ignite-Ember by AlanAyy (https://www.alanayy.com/)'''
```
Comment (AlphaRLee) [inline startup.py:34]: Better for a `contributors.md` file

```suggestion
    description='''Ignite-Ember by Schulich Ignite'''
```

**Pull request merged**